resources/publish/stage-maven-release.sh

```diff
@@ -60,43 +59,31 @@ while getopts ":ha:" option; do
   esac
 done
 
-[ -z "${SONATYPE_USERNAME}" ] && {
-  echo "SONATYPE_USERNAME is required"
-  exit 1
-}
-
-[ -z "${SONATYPE_PASSWORD}" ] && {
-  echo "SONATYPE_PASSWORD is required"
-  exit 1
-}
-
-[ -z "${JOB_NAME}" ] && {
-  echo "JOB_NAME is required"
-  exit 1
-}
-
-[ -z "${BUILD_ID}" ] && {
-  echo "BUILD_ID is required"
+if [ "$AUTO_PUBLISH" != "true" ] && [ "$AUTO_PUBLISH" != "false" ]; then
```

Comment:
default can be false, right?

Reply:
Yes https://github.com/opensearch-project/opensearch-build-libraries/pull/718/files#diff-4a19239627cfa172fa2404fa4a2feb02d26c4e26e1b2125e657d9f48ce4941edR41